doc/chassis/system-ports.md

```diff
@@ -0,0 +1,262 @@
+system_port_id   = 1*6DIGIT      ; globally unique port ID
+switch_id        = 1*2DIGIT      ; global switch ID. 
+core_id          = 1*2DIGIT      ; core id within switch_id. 
+core_port_id     = 1*6DIGIT      ; chip specific port  
+```
+
+The globally unique key in the SYSTEM_PORT_TABLE is the name of the ASIC instance and the front panel interface name. The ASIC name is chosen so that various agents like syncd can use this name as a filtering criterion to select the subset of entities that they need to operate on in order to manage a specific ASIC.
+
+The only restriction on selecting the ifname is that the key "SYSTEM_PORT|asic_name|ifname" must uniquely identify a port on the system.
+
+### 2.6.3 Inband Ports
+
+Inband ports are required to provide control plane connectivity between  forwarding engines. They are connected to the forwarding device local CPU on one side and the internal fabric on the other. 
+
+Every inband port is assigned a System Port ID just like front panel ports which is known to all the forwarding devices. Thus, every inband port is reachable from every forwarding engine.
+
+### 2.6.4 Fabric Ports
+
+The provisioning and management of Fabric ports is outside the scope of this document and will be documented as a separate proposal.
+
+## 2.7 Orchestration Agent
+
+### 2.7.1 System Port Handling
+
+System port configuration is expected to completely static and known at the start of the system.
+
+Based on the state of ‘“connect_chassis_db” in the device metadata in ConfigDB, Orchagent connects to Chassis DB and subscribes to the SYSTEM_PORT table in CHASSIS_DB. It uses this list of system ports from the SYSTEM_PORT table to construct the switch attributes needed by the create_switch SAI API. In a later phase, the system ports could be directly created by making sairedis calls from orchagent. 
+        
+A system port can be used in lieu of a physical port in several SAI API calls as relevant. For example, a system port can be added as a vlan member or be a lag member. To account for these, portsorch will be updated to support sysports. 
+
+
+Portsyncd does not have to support sysports because sysports do not have any associated kernel devices. 
+
+# 3 Testing
+
+Test coverage for the distributed VoQ architecture is achieved by extending the existing virtual switch based SWSS pytest infrastructure.
+
+The distributed switching architecture is represented as multiple VS instances connected with each other and called as Virtual Chassis, where one of the instance plays the role of the SSI and the remaining instances as FSIs. 
+
+## 3.1 Regression Testing
+
+Existing SWSS pytests can be executed against any of the instances in the virtual chassis to ensure that existing SONiC functionality is not affected while operating in a distributed environment.
+
+## 3.2 Distributed VoQ functionality
+
+Additional tests that specifically validate distributed VoQ forwarding functionality run only in the virtual chassis environment. 
```

Comment:
No plan as of now, since there is no SWSS behavior that is specific to the SSI. There may be some tests that exercise fabric device management that come along with the fabric management changes.